llama_stack/providers/inline/inference/meta_reference/inference.py

```diff
@@ -71,50 +70,66 @@ class MetaReferenceInferenceImpl(
 ):
     def __init__(self, config: MetaReferenceInferenceConfig) -> None:
         self.config = config
-        model = resolve_model(config.model)
-        if model is None:
-            raise RuntimeError(f"Unknown model: {config.model}, Run `llama model list`")
-        self.model_registry_helper = ModelRegistryHelper(
-            [
-                build_model_alias(
-                    model.descriptor(),
-                    model.core_model_id.value,
-                )
-            ],
-        )
-        self.model = model
-        # verify that the checkpoint actually is for this model lol
+        self.model_id = None
+        self.llama_model = None
 
-    async def initialize(self) -> None:
-        log.info(f"Loading model `{self.model.descriptor()}`")
+    async def initialize(self, model_id, llama_model) -> None:
```

Comment:
oh `initialize()` can never get an argument which is not specified in the config. the API is always:

```
# llama stack infra create a config for a provider 
# calls get_provider_impl(config)
# that method constructs the class 
# and initialize()s it
```